rock-paper-scissors/scripts/rock-paper-scissors.js

```diff
@@ -21,89 +41,110 @@ function playGame(playerSelection, computerSelection) {
     scissors(computerSelection);
   }
 }
-function computerPlay() {
-  let rPSS = ["rock", "paper", "scissors"];
-  let computerSelection = rPSS[Math.floor(Math.random() * rPSS.length)];
-  return computerSelection;
-}
-
-function roundCounter(roundCount) {}
 
 function rock(computerSelection) {
```

Comment:
There is a LOT of repetition going on between `rock()`, `paper()`, and `scissors()`. I'll put some general optimisations in th thread below.

Reply:
Instead of
```
roundCount += 1;
playerWins += 1;
```
being repeated in all three functions, you can write a function:
```
function calculatePlayerWins() {
    roundCount += 1;
    playerWins += 1;
}
```

Reply:
Adding the console.logs into the above functions, you'll need to add the `playerSelection` and `computerSelection` as variables.

So it becomes:
```
function calculatePlayerWins(playerSelection, computerSelection) {
    roundCount += 1;
    playerWins += 1;
    console.log(`You win! ${playerSelection} beats ${computerSelection}`);
}
```

Reply:
Once you've got all 3 functions:
```
function calculateNobodyWins() {
    roundCount += 1;
    nobodyWins += 1;
    console.log(`This round was a draw! Try again!`);
}

function calculatePlayerWins(playerSelection, computerSelection) {
    roundCount += 1;
    playerWins += 1;
    console.log(`You win! ${playerSelection} beats ${computerSelection}`);
}

function calculateComputerWins(playerSelection, computerSelection) {
    roundCount += 1;
    computerWins += 1;
    console.log(``You Lose! ${computerSelection} beats ${playerSelection}!`);
}
```

There is one more (obvious) point of repetition. Can you spot it? It's to do with `roundCount`.